src/main/java/net/dries007/tfc/util/events/StartFireEvent.java

```diff
@@ -32,12 +49,32 @@
  */
 public final class StartFireEvent extends Event implements ICancellableEvent
 {
+    // Replicate vanilla fire starting sounds that do not fire
+    public static boolean startFireWithSound(Level level, BlockPos pos, BlockState state, Direction direction, @Nullable Player player, ItemStack stack)
+    {
+        final boolean startFire = startFire(level, pos, state, direction, player, stack, FireStrength.STRONG, 1.0);
+        if (startFire)
+        {
+            if (stack.getItem() instanceof FlintAndSteelItem)
+            {
+                level.playSound(player, pos, SoundEvents.FLINTANDSTEEL_USE, SoundSource.BLOCKS, 1.0F, level.getRandom().nextFloat() * 0.4F + 0.8F);
+            }
+            else if (stack.getItem() instanceof FireChargeItem)
+            {
+                level.playSound(null, pos, SoundEvents.FIRECHARGE_USE, SoundSource.BLOCKS, 1.0F, (level.getRandom().nextFloat() - level.getRandom().nextFloat()) * 0.2F + 1.0F);
+            }
+
+        }
+        return startFire;
+    }
+
     public static boolean startFire(Level level, BlockPos pos, BlockState state, Direction direction, @Nullable Player player, ItemStack stack)
     {
-        return startFire(level, pos, state, direction, player, stack, FireStrength.STRONG);
+        return startFire(level, pos, state, direction, player, stack, FireStrength.STRONG, 1.0);
     }
 
-    public static boolean startFire(Level level, BlockPos pos, BlockState state, Direction direction, @Nullable Player player, ItemStack stack, FireStrength strength)
+    // Pass in a firepitBaseChance of -1 to disable firepit creation for a given firestarter
+    public static boolean startFire(Level level, BlockPos pos, BlockState state, Direction direction, @Nullable Player player, ItemStack stack, FireStrength strength, double firepitBaseChance)
```

Comment:
I am not sure if I love firepit chance getting its own parameter for the firestarting event... but I guess I see the use case.